daemon/src/listen.rs

```diff
@@ -0,0 +1,118 @@
+use std::{
+    io::{self, BufRead, BufReader, BufWriter, Write},
+    os::unix::{
+        net::{UnixListener, UnixStream},
+        prelude::PermissionsExt,
+    },
+    thread,
+};
+
+use common::{
+    client_daemon::{
+        parse_message_client_to_daemon, serialize_message_daemon_to_client, ClientToDaemonMsg,
+        DaemonToClientMsg,
+    },
+    notif::notif,
+};
+
+pub fn listen(path: &'static str) {
+    thread::spawn(move || {
+        // Get rid of the old sock
+        std::fs::remove_file(path).ok();
+
+        // Try to handle sock connections then
+        let listener = match UnixListener::bind(path) {
+            Ok(listener) => listener,
+            Err(e) => {
+                notif(
+                    "Daemon Error",
+                    &format!("Failed to bind to {}: {}", path, e),
+                );
+                return;
+            }
+        };
+
+        // Set the permissions on the sock
+        std::fs::set_permissions(path, std::fs::Permissions::from_mode(0o777)).ok();
```

Comment:
This lets other users read messages on accounts that aren't theirs. Maybe remove the groups perm.

Reply:
Good idea!